Installation issue: gromacs

### Steps to reproduce the issue

```console
$ spack spec -I --fresh gromacs@main %gcc@13 +sycl ^hipsycl +rocm amdgpu_target=gfx1100 ^openmpi fabrics=auto
Input spec
--------------------------------
 -   gromacs@main%gcc@13+sycl
 -       ^hipsycl+rocm amdgpu_target=gfx1100
 -       ^openmpi fabrics=auto

Concretized
--------------------------------
 -   gromacs@main%gcc@13.3.0~cp2k~cuda~cycle_subcounters~double+gmxapi~heffte+hwloc~intel-data-center-gpu-max~intel_provided_gcc~ipo~mdrun_only+mpi+nblib~nosuffix~opencl+openmp~relaxed_double_precision+shared+sycl build_system=cmake build_type=Release generator=make openmp_max_threads=none arch=linux-debian12-zen4
[+]      ^cmake@3.30.5%gcc@13.3.0~doc+ncurses+ownlibs~qtgui build_system=generic build_type=Release patches=dbc3892 arch=linux-debian12-zen4
[+]          ^curl@8.10.1%gcc@13.3.0~gssapi~ldap~libidn2~librtmp~libssh~libssh2+nghttp2 build_system=autotools libs=shared,static tls=openssl arch=linux-debian12-zen4
[+]              ^nghttp2@1.63.0%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                  ^diffutils@3.10%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]              ^openssl@3.3.1%gcc@13.3.0~docs+shared build_system=generic certs=mozilla arch=linux-debian12-zen4
[-]                  ^ca-certificates-mozilla@2023-05-30%gcc@13.3.0 build_system=generic arch=linux-debian12-zen4
[+]          ^ncurses@6.5%gcc@13.3.0~symlinks+termlib abi=none build_system=autotools patches=7a351bc arch=linux-debian12-zen4
[+]          ^zlib-ng@2.2.1%gcc@13.3.0+compat+new_strategies+opt+pic+shared build_system=autotools arch=linux-debian12-zen4
[+]      ^fftw@3.3.10%gcc@13.3.0+mpi~openmp~pfft_patches+shared build_system=autotools patches=872cff9 precision=double,float arch=linux-debian12-zen4
[+]      ^gcc-runtime@13.3.0%gcc@13.3.0 build_system=generic arch=linux-debian12-zen4
[e]      ^glibc@2.36%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]      ^gmake@4.4.1%gcc@13.3.0~guile build_system=generic arch=linux-debian12-zen4
[+]      ^hipsycl@24.06.0%gcc@13.3.0~cuda~ipo+rocm amdgpu_target=gfx1100 build_system=cmake build_type=Release generator=make arch=linux-debian12-zen4
[+]          ^boost@1.69.0%gcc@13.3.0~atomic~chrono~clanglibcpp~container+context~contract~coroutine~date_time~debug~exception+fiber+filesystem~graph~graph_parallel~icu~iostreams~json~locale~log~math~mpi+multithreaded~nowide~numpy~pic~program_options~python~random~regex~serialization+shared~signals~singlethreaded~stacktrace~system~taggedlayout~test~thread~timer~type_erasure~url~versionedlayout~wave build_system=generic context-impl=fcontext cxxstd=17 patches=3a83d90,5da7ad2,607b077 visibility=hidden arch=linux-debian12-zen4
[+]          ^hip@6.2.1%gcc@13.3.0~asan~cuda~ipo+rocm build_system=cmake build_type=Release generator=make patches=1f65dfe arch=linux-debian12-zen4
[+]              ^comgr@6.2.1%gcc@13.3.0~asan~ipo build_system=cmake build_type=Release generator=make arch=linux-debian12-zen4
[+]                  ^rocm-cmake@6.2.1%gcc@13.3.0~ipo build_system=cmake build_type=Release generator=make arch=linux-debian12-zen4
[+]              ^glx@1.4%gcc@13.3.0 build_system=bundle arch=linux-debian12-zen4
[+]                  ^mesa@23.3.6%gcc@13.3.0+glx+llvm+opengl~opengles+osmesa~strip build_system=meson buildtype=release default_library=shared arch=linux-debian12-zen4
[+]                      ^flex@2.6.3%gcc@13.3.0+lex~nls build_system=autotools arch=linux-debian12-zen4
[+]                      ^glproto@1.4.17%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                      ^libunwind@1.6.2%gcc@13.3.0~block_signals~conservative_checks~cxx_exceptions~debug~debug_frame+docs~pic+tests+weak_backtrace~xz~zlib build_system=autotools components=none libs=shared,static arch=linux-debian12-zen4
[+]                      ^libx11@1.8.10%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                          ^inputproto@2.3.2%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                          ^kbproto@1.0.7%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                          ^xextproto@7.3.0%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                          ^xproto@7.0.31%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                          ^xtrans@1.5.0%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                      ^libxcb@1.17.0%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                          ^libxau@1.0.11%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                          ^libxdmcp@1.1.5%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                          ^xcb-proto@1.17.0%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                      ^libxext@1.3.6%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                      ^libxt@1.3.0%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                          ^libice@1.1.1%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                          ^libsm@1.2.4%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                      ^meson@1.5.1%gcc@13.3.0 build_system=python_pip patches=0f0b1bd arch=linux-debian12-zen4
[+]                      ^py-mako@1.2.4%gcc@13.3.0 build_system=python_pip arch=linux-debian12-zen4
[+]                          ^py-markupsafe@2.1.3%gcc@13.3.0 build_system=python_pip arch=linux-debian12-zen4
[+]                      ^xrandr@1.5.2%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                          ^libxrandr@1.5.4%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                              ^renderproto@0.11.1%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                          ^libxrender@0.9.11%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                          ^randrproto@1.5.0%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]              ^hipcc@6.2.1%gcc@13.3.0~ipo build_system=cmake build_type=Release generator=make patches=c10b010 arch=linux-debian12-zen4
[+]              ^hipify-clang@6.2.1%gcc@13.3.0~asan~ipo build_system=cmake build_type=Release generator=make patches=16e0e2b arch=linux-debian12-zen4
[+]              ^hsakmt-roct@6.2.1%gcc@13.3.0~asan~ipo+shared build_system=cmake build_type=Release generator=make arch=linux-debian12-zen4
[+]                  ^libdrm@2.4.123%gcc@13.3.0~docs~strip build_system=meson buildtype=release default_library=shared arch=linux-debian12-zen4
[+]                      ^libpthread-stubs@0.5%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]              ^libedit@3.1-20240808%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]              ^perl-file-which@1.27%gcc@13.3.0 build_system=perl arch=linux-debian12-zen4
[+]              ^perl-uri-encode@1.1.1%gcc@13.3.0 build_system=perl arch=linux-debian12-zen4
[+]                  ^perl-module-build@0.4234%gcc@13.3.0 build_system=perl arch=linux-debian12-zen4
[+]              ^py-cppheaderparser@2.7.4%gcc@13.3.0 build_system=python_pip arch=linux-debian12-zen4
[+]                  ^py-pip@23.1.2%gcc@13.3.0 build_system=generic arch=linux-debian12-zen4
[+]                  ^py-ply@3.11%gcc@13.3.0 build_system=python_pip arch=linux-debian12-zen4
[+]                  ^py-setuptools@69.2.0%gcc@13.3.0 build_system=generic arch=linux-debian12-zen4
[+]                  ^py-wheel@0.41.2%gcc@13.3.0 build_system=generic arch=linux-debian12-zen4
[+]                  ^python-venv@1.0%gcc@13.3.0 build_system=generic arch=linux-debian12-zen4
[+]              ^rocm-core@6.2.1%gcc@13.3.0~asan~ipo build_system=cmake build_type=Release generator=make arch=linux-debian12-zen4
[+]              ^rocminfo@6.2.1%gcc@13.3.0~ipo build_system=cmake build_type=Release generator=make arch=linux-debian12-zen4
[+]              ^rocprofiler-register@6.2.1%gcc@13.3.0~ipo build_system=cmake build_type=Release generator=make patches=fc2f3cd arch=linux-debian12-zen4
[+]                  ^fmt@11.0.2%gcc@13.3.0~ipo+pic~shared build_system=cmake build_type=Release cxxstd=11 generator=make arch=linux-debian12-zen4
[+]                  ^glog@0.7.1%gcc@13.3.0~ipo build_system=cmake build_type=Release generator=make arch=linux-debian12-zen4
[+]                      ^gflags@2.2.2%gcc@13.3.0~ipo build_system=cmake build_type=Release generator=make arch=linux-debian12-zen4
[+]              ^roctracer-dev-api@6.2.1%gcc@13.3.0 build_system=generic arch=linux-debian12-zen4
[+]          ^hsa-rocr-dev@6.2.1%gcc@13.3.0~asan+image~ipo+shared build_system=cmake build_type=Release generator=make arch=linux-debian12-zen4
[+]              ^elfutils@0.191%gcc@13.3.0~debuginfod+exeprefix+nls build_system=autotools arch=linux-debian12-zen4
[+]                  ^zstd@1.5.6%gcc@13.3.0+programs build_system=makefile compression=none libs=shared,static arch=linux-debian12-zen4
[+]              ^xxd-standalone@8.2.1201%gcc@13.3.0 build_system=makefile arch=linux-debian12-zen4
[+]          ^llvm@17.0.6%gcc@13.3.0+clang~cuda~flang+gold~ipo+libomptarget~libomptarget_debug~link_llvm_dylib+lld+lldb+llvm_dylib+lua~mlir+polly~python~split_dwarf~z3~zstd build_system=cmake build_type=Release compiler-rt=runtime generator=ninja libcxx=runtime libunwind=runtime openmp=runtime shlib_symbol_version=none targets=all version_suffix=none arch=linux-debian12-zen4
[+]              ^binutils@2.43.1%gcc@13.3.0~gas+gold~gprofng+headers~interwork+ld~libiberty~lto~nls~pgo+plugins build_system=autotools compress_debug_sections=zlib libs=shared,static arch=linux-debian12-zen4
[+]              ^libffi@3.4.6%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]              ^lua@5.3.6%gcc@13.3.0~pcfile+shared build_system=makefile fetcher=curl arch=linux-debian12-zen4
[+]                  ^unzip@6.0%gcc@13.3.0 build_system=makefile patches=881d2ed,f6f6236 arch=linux-debian12-zen4
[+]              ^ninja@1.12.1%gcc@13.3.0+re2c build_system=generic arch=linux-debian12-zen4
[+]                  ^re2c@3.1%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]              ^perl-data-dumper@2.173%gcc@13.3.0 build_system=perl arch=linux-debian12-zen4
[+]              ^swig@4.1.1%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                  ^pcre2@10.44%gcc@13.3.0~jit+multibyte build_system=autotools arch=linux-debian12-zen4
[+]              ^xz@5.4.6%gcc@13.3.0~pic build_system=autotools libs=shared,static arch=linux-debian12-zen4
[+]          ^llvm-amdgpu@6.2.1%gcc@13.3.0~ipo~link_llvm_dylib~llvm_dylib+rocm-device-libs build_system=cmake build_type=Release generator=ninja patches=b4774ca,eaf700a arch=linux-debian12-zen4
[+]              ^z3@4.12.4%gcc@13.3.0~gmp~ipo~python build_system=cmake build_type=Release generator=make arch=linux-debian12-zen4
[+]          ^python@3.11.9%gcc@13.3.0+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=13fa8bf,b0615b2,ebdca64,f2fd060 arch=linux-debian12-zen4
[+]              ^bzip2@1.0.8%gcc@13.3.0~debug~pic+shared build_system=generic arch=linux-debian12-zen4
[+]              ^expat@2.6.3%gcc@13.3.0+libbsd build_system=autotools arch=linux-debian12-zen4
[+]                  ^libbsd@0.12.2%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                      ^libmd@1.0.4%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]              ^gdbm@1.23%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]              ^gettext@0.22.5%gcc@13.3.0+bzip2+curses+git~libunistring+libxml2+pic+shared+tar+xz build_system=autotools arch=linux-debian12-zen4
[+]                  ^tar@1.34%gcc@13.3.0 build_system=autotools zip=pigz arch=linux-debian12-zen4
[+]                      ^pigz@2.8%gcc@13.3.0 build_system=makefile arch=linux-debian12-zen4
[+]              ^libxcrypt@4.4.35%gcc@13.3.0~obsolete_api build_system=autotools patches=4885da3 arch=linux-debian12-zen4
[+]              ^readline@8.2%gcc@13.3.0 build_system=autotools patches=bbf97f1 arch=linux-debian12-zen4
[+]              ^sqlite@3.46.0%gcc@13.3.0+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=linux-debian12-zen4
[+]              ^util-linux-uuid@2.40.2%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]      ^hwloc@2.11.1%gcc@13.3.0~cairo~cuda~gl~libudev+libxml2~nvml~oneapi-level-zero~opencl+pci~rocm build_system=autotools libs=shared,static arch=linux-debian12-zen4
[+]          ^libpciaccess@0.17%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]              ^util-macros@1.20.1%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]          ^libxml2@2.10.3%gcc@13.3.0+pic~python+shared build_system=autotools arch=linux-debian12-zen4
[+]              ^libiconv@1.17%gcc@13.3.0 build_system=autotools libs=shared,static arch=linux-debian12-zen4
[+]      ^openblas@0.3.28%gcc@13.3.0~bignuma~consistent_fpcsr+dynamic_dispatch+fortran~ilp64+locking+pic+shared build_system=makefile patches=d0b9276 symbol_suffix=none threads=none arch=linux-debian12-zen4
[+]      ^openmpi@5.0.5%gcc@13.3.0+atomics~cuda~debug~gpfs~internal-hwloc~internal-libevent~internal-pmix~java~lustre~memchecker~openshmem~romio+rsh~static+vt+wrapper-rpath build_system=autotools fabrics=auto romio-filesystem=none schedulers=none arch=linux-debian12-zen4
[+]          ^autoconf@2.72%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]              ^m4@1.4.19%gcc@13.3.0+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-debian12-zen4
[+]                  ^libsigsegv@2.14%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]          ^automake@1.16.5%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]          ^libevent@2.1.12%gcc@13.3.0+openssl build_system=autotools arch=linux-debian12-zen4
[+]          ^libtool@2.4.7%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]              ^findutils@4.9.0%gcc@13.3.0 build_system=autotools patches=440b954 arch=linux-debian12-zen4
[+]          ^numactl@2.0.18%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]          ^openssh@9.8p1%gcc@13.3.0+gssapi build_system=autotools arch=linux-debian12-zen4
[+]              ^krb5@1.21.3%gcc@13.3.0+shared build_system=autotools arch=linux-debian12-zen4
[+]                  ^bison@3.8.2%gcc@13.3.0~color build_system=autotools arch=linux-debian12-zen4
[+]          ^perl@5.40.0%gcc@13.3.0+cpanm+opcode+open+shared+threads build_system=generic arch=linux-debian12-zen4
[+]              ^berkeley-db@18.1.40%gcc@13.3.0+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-debian12-zen4
[+]          ^pmix@5.0.3%gcc@13.3.0~munge~python~restful build_system=autotools arch=linux-debian12-zen4
[+]      ^pkgconf@2.2.0%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
```


### Error message

<details><summary>Error message</summary>

<pre>
1 error found in build log:
     158    -- Performing Test _callconv___vectorcall
     159    -- Performing Test _callconv___vectorcall - Failed
     160    -- Performing Test _callconv___regcall
     161    -- Performing Test _callconv___regcall - Failed
     162    -- Performing Test _callconv_
     163    -- Performing Test _callconv_  - Success
  >> 164    CMake Error at cmake/gmxManageSyclAdaptiveCpp.cmake:66 (message):
     165      AdaptiveCpp/hipSYCL build requires Clang compiler, but GNU is used
     166    Call Stack (most recent call first):
     167      cmake/gmxManageSycl.cmake:45 (include)
     168      CMakeLists.txt:718 (include)
     169
     170
</pre></details>


### Information on your system

* **Spack:** 0.23.0.dev0 (fea2171672737687195572d920cc911d3c194229)
* **Python:** 3.11.2
* **Platform:** linux-debian12-zen4

### Additional information

[spack-build-env.txt](https://github.com/user-attachments/files/17547822/spack-build-env.txt)
[spack-build-out.txt](https://github.com/user-attachments/files/17547823/spack-build-out.txt)

@eirrgang @junghans  @mabraham

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

> See #47274

thanks for your commit, after merging this commit, it gives

```
$ spack install --fresh gromacs@main %gcc@13 +sycl ^hipsycl +rocm amdgpu_target=gfx1100 ^openmpi fabrics=auto
==> Error: Spack concretizer internal error. Please submit a bug report and include the command, environment if applicable and the following error message.
    gromacs@main%gcc@13+sycl ^hipsycl+rocm amdgpu_target=gfx1100 ^openmpi fabrics=auto is unsatisfiable
```

if ommit the ```+sycl```, the installation could continue (should this variant add or not?), 


<details><summary>spack spec</summary>
<pre>
$ spack spec -I --fresh gromacs@main %gcc@13 ^hipsycl +rocm amdgpu_target=gfx1100 ^openmpi fabrics=auto
Input spec
--------------------------------
 -   gromacs@main%gcc@13
 -       ^hipsycl+rocm amdgpu_target=gfx1100
 -       ^openmpi fabrics=auto

Concretized
--------------------------------
 -   gromacs@main%gcc@13.3.0+cp2k~cuda~cycle_subcounters~double+gmxapi+hwloc~intel_provided_gcc~ipo~mdrun_only+mpi+nblib~nosuffix~opencl+openmp~relaxed_double_precision~shared build_system=cmake build_type=Release generator=make openmp_max_threads=none arch=linux-debian12-zen4
[+]      ^cmake@3.30.5%gcc@13.3.0~doc+ncurses+ownlibs~qtgui build_system=generic build_type=Release patches=dbc3892 arch=linux-debian12-zen4
[+]          ^curl@8.10.1%gcc@13.3.0~gssapi~ldap~libidn2~librtmp~libssh~libssh2+nghttp2 build_system=autotools libs=shared,static tls=openssl arch=linux-debian12-zen4
[+]              ^nghttp2@1.63.0%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                  ^diffutils@3.10%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]              ^openssl@3.3.1%gcc@13.3.0~docs+shared build_system=generic certs=mozilla arch=linux-debian12-zen4
[-]                  ^ca-certificates-mozilla@2023-05-30%gcc@13.3.0 build_system=generic arch=linux-debian12-zen4
[+]          ^ncurses@6.5%gcc@13.3.0~symlinks+termlib abi=none build_system=autotools patches=7a351bc arch=linux-debian12-zen4
[+]          ^zlib-ng@2.2.1%gcc@13.3.0+compat+new_strategies+opt+pic+shared build_system=autotools arch=linux-debian12-zen4
 -       ^cp2k@2024.3%gcc@13.3.0~cosma~cuda+dlaf~elpa~enable_regtests~ipo+libint~libvori+libxc+mpi~mpi_f08+openmp~pexsi~plumed~pytorch~quip~rocm~sirius~spglib~spla build_system=cmake build_type=Release generator=make lmax=5 patches=37f4f1a,cf8e9cf smm=libxsmm arch=linux-debian12-zen4
[+]          ^dbcsr@2.7.0%gcc@13.3.0~cuda~cuda_arch_35_k20x~examples~g2g~ipo+mpi~mpi_f08~opencl+openmp~rocm+shared build_system=cmake build_type=Release generator=ninja smm=libxsmm arch=linux-debian12-zen4
[+]              ^ninja@1.12.1%gcc@13.3.0+re2c build_system=generic arch=linux-debian12-zen4
[+]                  ^re2c@3.1%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
 -           ^dla-future@0.6.0%gcc@13.3.0~cuda~doc~hdf5~ipo~miniapps~mpi_gpu_aware~rocm+scalapack+shared build_system=cmake build_type=Release generator=ninja arch=linux-debian12-zen4
[+]              ^blaspp@2024.05.31%gcc@13.3.0~cuda~ipo+openmp~rocm+shared~sycl build_system=cmake build_type=Release generator=make arch=linux-debian12-zen4
[+]              ^lapackpp@2024.05.31%gcc@13.3.0~cuda~ipo~rocm+shared~sycl build_system=cmake build_type=Release generator=make arch=linux-debian12-zen4
 -               ^pika@0.27.0%gcc@13.3.0+apex~cuda~examples~generic_coroutines~ipo+mpi~rocm~sanitizers+shared~stdexec~tracy~valgrind build_system=cmake build_type=Release cxxstd=17 generator=ninja malloc=mimalloc arch=linux-debian12-zen4
 -                   ^apex@2.6.5%aocc@5.0.0~activeharmony+binutils~cuda~examples~gperftools~hip~ipo~jemalloc+kokkos~lmsensors~mpi+openmp+otf2+papi~plugins~starpu+sycl~tests build_system=cmake build_type=Release generator=make arch=linux-debian12-zen4
[+]                      ^gmake@4.4.1%aocc@5.0.0~guile build_system=generic arch=linux-debian12-zen4
 -                       ^hipsycl@0.8.0%gcc@13.3.0~cuda~ipo+rocm amdgpu_target=gfx1100 build_system=cmake build_type=Release generator=make arch=linux-debian12-zen4
 -                           ^hip@6.2.1%gcc@13.3.0~asan~cuda~ipo+rocm build_system=cmake build_type=Release generator=make patches=1f65dfe arch=linux-debian12-zen4
[+]                              ^comgr@6.2.1%gcc@13.3.0~asan~ipo build_system=cmake build_type=Release generator=make arch=linux-debian12-zen4
[+]                                  ^rocm-cmake@6.2.1%gcc@13.3.0~ipo build_system=cmake build_type=Release generator=make arch=linux-debian12-zen4
 -                               ^glx@1.4%gcc@13.3.0 build_system=bundle arch=linux-debian12-zen4
 -                                   ^mesa@23.3.6%gcc@13.3.0+glx+llvm+opengl~opengles+osmesa~strip build_system=meson buildtype=release default_library=shared arch=linux-debian12-zen4
[+]                                      ^flex@2.6.3%gcc@13.3.0+lex~nls build_system=autotools arch=linux-debian12-zen4
[+]                                      ^glproto@1.4.17%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                                      ^libunwind@1.6.2%gcc@13.3.0~block_signals~conservative_checks~cxx_exceptions~debug~debug_frame+docs~pic+tests+weak_backtrace~xz~zlib build_system=autotools components=none libs=shared,static arch=linux-debian12-zen4
[+]                                      ^libx11@1.8.10%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                                          ^inputproto@2.3.2%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                                          ^kbproto@1.0.7%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                                          ^xextproto@7.3.0%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                                          ^xproto@7.0.31%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                                          ^xtrans@1.5.0%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                                      ^libxcb@1.17.0%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                                          ^libxau@1.0.11%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                                          ^libxdmcp@1.1.5%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                                          ^xcb-proto@1.17.0%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                                      ^libxext@1.3.6%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                                      ^libxt@1.3.0%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                                          ^libice@1.1.1%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                                          ^libsm@1.2.4%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                                      ^meson@1.5.1%gcc@13.3.0 build_system=python_pip patches=0f0b1bd arch=linux-debian12-zen4
[+]                                      ^py-mako@1.2.4%gcc@13.3.0 build_system=python_pip arch=linux-debian12-zen4
[+]                                          ^py-markupsafe@2.1.3%gcc@13.3.0 build_system=python_pip arch=linux-debian12-zen4
[+]                                      ^xrandr@1.5.2%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                                          ^libxrandr@1.5.4%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                                              ^renderproto@0.11.1%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                                          ^libxrender@0.9.11%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                                          ^randrproto@1.5.0%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                              ^hipcc@6.2.1%gcc@13.3.0~ipo build_system=cmake build_type=Release generator=make patches=c10b010 arch=linux-debian12-zen4
[+]                              ^hipify-clang@6.2.1%gcc@13.3.0~asan~ipo build_system=cmake build_type=Release generator=make patches=16e0e2b arch=linux-debian12-zen4
[+]                              ^hsakmt-roct@6.2.1%gcc@13.3.0~asan~ipo+shared build_system=cmake build_type=Release generator=make arch=linux-debian12-zen4
[+]                                  ^libdrm@2.4.123%gcc@13.3.0~docs~strip build_system=meson buildtype=release default_library=shared arch=linux-debian12-zen4
[+]                                      ^libpthread-stubs@0.5%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                              ^perl-file-which@1.27%gcc@13.3.0 build_system=perl arch=linux-debian12-zen4
[+]                              ^perl-uri-encode@1.1.1%gcc@13.3.0 build_system=perl arch=linux-debian12-zen4
[+]                                  ^perl-module-build@0.4234%gcc@13.3.0 build_system=perl arch=linux-debian12-zen4
[+]                              ^py-cppheaderparser@2.7.4%gcc@13.3.0 build_system=python_pip arch=linux-debian12-zen4
[+]                                  ^py-ply@3.11%gcc@13.3.0 build_system=python_pip arch=linux-debian12-zen4
[+]                              ^rocm-core@6.2.1%gcc@13.3.0~asan~ipo build_system=cmake build_type=Release generator=make arch=linux-debian12-zen4
 -                               ^rocminfo@6.2.1%gcc@13.3.0~ipo build_system=cmake build_type=Release generator=make arch=linux-debian12-zen4
[+]                              ^rocprofiler-register@6.2.1%gcc@13.3.0~ipo build_system=cmake build_type=Release generator=make patches=fc2f3cd arch=linux-debian12-zen4
[+]                                  ^glog@0.7.1%gcc@13.3.0~ipo build_system=cmake build_type=Release generator=make arch=linux-debian12-zen4
[+]                                      ^gflags@2.2.2%gcc@13.3.0~ipo build_system=cmake build_type=Release generator=make arch=linux-debian12-zen4
[+]                              ^roctracer-dev-api@6.2.1%gcc@13.3.0 build_system=generic arch=linux-debian12-zen4
 -                           ^hsa-rocr-dev@6.2.1%gcc@13.3.0~asan+image~ipo+shared build_system=cmake build_type=Release generator=make arch=linux-debian12-zen4
[+]                              ^elfutils@0.191%gcc@13.3.0~debuginfod+exeprefix+nls build_system=autotools arch=linux-debian12-zen4
[+]                              ^xxd-standalone@8.2.1201%gcc@13.3.0 build_system=makefile arch=linux-debian12-zen4
 -                           ^llvm@10.0.1%gcc@13.3.0+clang~cuda~flang+gold~ipo~libomp_tsan+libomptarget~libomptarget_debug~link_llvm_dylib+lld+lldb+llvm_dylib~lua~mlir+polly~python~split_dwarf~z3 build_system=cmake build_type=Release compiler-rt=runtime generator=ninja libcxx=runtime libunwind=runtime openmp=project patches=0e121ed,50bfc4e,514926d,66932ba,6f0cfa5,7425017,c6ca6b9 targets=all version_suffix=none arch=linux-debian12-zen4
[+]                              ^cmake@3.16.9%gcc@13.3.0~doc+ncurses+ownlibs~qtgui build_system=generic build_type=Release patches=1c54004,bf695e3 arch=linux-debian12-zen4
[+]                              ^perl-data-dumper@2.173%gcc@13.3.0 build_system=perl arch=linux-debian12-zen4
[+]                          ^llvm-amdgpu@6.2.1%gcc@13.3.0~ipo~link_llvm_dylib~llvm_dylib+rocm-device-libs build_system=cmake build_type=Release generator=ninja patches=b4774ca,eaf700a arch=linux-debian12-zen4
[+]                              ^z3@4.12.4%gcc@13.3.0~gmp~ipo~python build_system=cmake build_type=Release generator=make arch=linux-debian12-zen4
[+]                      ^kokkos@4.4.01%gcc@13.3.0~aggressive_vectorization~compiler_warnings~cuda~debug~debug_bounds_check~debug_dualview_modify_check~deprecated_code~examples~hpx~hpx_async_dispatch~hwloc~ipo~memkind~numactl~openmp~openmptarget~pic~rocm+serial+shared~sycl~tests~threads~tuning~wrapper build_system=cmake build_type=Release cxxstd=17 generator=make intel_gpu_arch=none arch=linux-debian12-zen4
[+]                      ^otf2@3.0.3%gcc@13.3.0 build_system=autotools patches=7e56d93 arch=linux-debian12-zen4
[+]                      ^papi@7.1.0%aocc@5.0.0~cuda~debug+example~infiniband~lmsensors~nvml~powercap~rapl+rdpmc~rocm~rocm_smi~sde+shared~static_tools build_system=autotools patches=48cb202 arch=linux-debian12-zen4
[+]                  ^fmt@11.0.2%gcc@13.3.0~ipo+pic~shared build_system=cmake build_type=Release cxxstd=17 generator=make arch=linux-debian12-zen4
[+]                  ^mimalloc@2.1.7%gcc@13.3.0~build_tests~debug_full~ipo~local_dynamic_tls+override+padding~secure~see_asm~show_errors~skip_collect_on_exit~use_cxx~xmalloc build_system=cmake build_type=Release generator=make libs=object,shared,static arch=linux-debian12-zen4
[+]                  ^spdlog@1.14.1%gcc@13.3.0~ipo+shared build_system=cmake build_type=Release generator=make arch=linux-debian12-zen4
[+]              ^umpire@2024.07.0%gcc@13.3.0~asan~backtrace+c~cuda~dev_benchmarks~device_alloc~deviceconst~examples+fmt_header_only~fortran~ipc_shmem~ipo~mpi~numa~omptarget~openmp~rocm~sanitizer_tests+shared~sqlite_experimental~tools~werror build_system=cmake build_type=Release generator=make tests=none arch=linux-debian12-zen4
[+]                  ^blt@0.6.2%gcc@13.3.0 build_system=generic arch=linux-debian12-zen4
[+]                  ^camp@2024.07.0%gcc@13.3.0~cuda~ipo~omptarget~openmp~rocm~sycl~tests build_system=cmake build_type=Release generator=make arch=linux-debian12-zen4
 -           ^dla-future-fortran@0.2.0%gcc@13.3.0~ipo+shared~test build_system=cmake build_type=Release generator=ninja arch=linux-debian12-zen4
[+]          ^libint@2.9.0%gcc@13.3.0~debug~fma+fortran build_system=autotools tune=cp2k-lmax-5 arch=linux-debian12-zen4
[+]              ^boost@1.86.0%gcc@13.3.0+atomic~charconv+chrono~clanglibcpp~cobalt~container~context~contract~coroutine+date_time~debug+exception~fiber+filesystem+graph~graph_parallel~icu+iostreams~json+locale+log+math~mpi+multithreaded~nowide~numpy~pic+program_options~python+random+regex+serialization+shared+signals~singlethreaded~stacktrace+system~taggedlayout+test+thread+timer~type_erasure~url~versionedlayout+wave build_system=generic cxxstd=17 patches=a440f96 visibility=hidden arch=linux-debian12-zen4
[+]                  ^zstd@1.5.6%gcc@13.3.0+programs build_system=makefile compression=none libs=shared,static arch=linux-debian12-zen4
[+]              ^eigen@3.4.0%gcc@13.3.0~ipo build_system=cmake build_type=RelWithDebInfo generator=make arch=linux-debian12-zen4
[+]              ^gmp@6.3.0%gcc@13.3.0+cxx build_system=autotools libs=shared,static arch=linux-debian12-zen4
[+]          ^libxc@6.2.2%gcc@13.3.0~cuda~kxc~lxc+shared build_system=autotools arch=linux-debian12-zen4
[+]          ^libxsmm@1.17%gcc@13.3.0~debug~generator~header-only~large_jit_buffer~shared blas=default build_system=makefile arch=linux-debian12-zen4
[+]              ^binutils@2.43.1%gcc@13.3.0+gas+gold~gprofng+headers~interwork+ld+libiberty~lto~nls~pgo+plugins build_system=autotools compress_debug_sections=zlib libs=shared,static arch=linux-debian12-zen4
[+]          ^netlib-scalapack@2.2.0%gcc@13.3.0~ipo~pic+shared build_system=cmake build_type=Release generator=make patches=072b006,1c9ce5f,244a9aa arch=linux-debian12-zen4
[+]          ^py-fypp@3.1%gcc@13.3.0 build_system=python_pip arch=linux-debian12-zen4
[+]              ^py-pip@23.1.2%gcc@13.3.0 build_system=generic arch=linux-debian12-zen4
[+]              ^py-setuptools@69.2.0%gcc@13.3.0 build_system=generic arch=linux-debian12-zen4
[+]              ^py-wheel@0.41.2%gcc@13.3.0 build_system=generic arch=linux-debian12-zen4
[+]              ^python-venv@1.0%gcc@13.3.0 build_system=generic arch=linux-debian12-zen4
[+]          ^python@3.11.9%gcc@13.3.0+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=13fa8bf,b0615b2,ebdca64,f2fd060 arch=linux-debian12-zen4
[+]              ^bzip2@1.0.8%gcc@13.3.0~debug~pic+shared build_system=generic arch=linux-debian12-zen4
[+]              ^expat@2.6.3%gcc@13.3.0+libbsd build_system=autotools arch=linux-debian12-zen4
[+]                  ^libbsd@0.12.2%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]                      ^libmd@1.0.4%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]              ^gdbm@1.23%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]              ^gettext@0.22.5%gcc@13.3.0+bzip2+curses+git~libunistring+libxml2+pic+shared+tar+xz build_system=autotools arch=linux-debian12-zen4
[+]                  ^tar@1.34%gcc@13.3.0 build_system=autotools zip=pigz arch=linux-debian12-zen4
[+]                      ^pigz@2.8%gcc@13.3.0 build_system=makefile arch=linux-debian12-zen4
[+]              ^libffi@3.4.6%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]              ^libxcrypt@4.4.35%gcc@13.3.0~obsolete_api build_system=autotools patches=4885da3 arch=linux-debian12-zen4
[+]              ^readline@8.2%gcc@13.3.0 build_system=autotools patches=bbf97f1 arch=linux-debian12-zen4
[+]              ^sqlite@3.46.0%gcc@13.3.0+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=linux-debian12-zen4
[+]              ^util-linux-uuid@2.40.2%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]              ^xz@5.4.6%gcc@13.3.0~pic build_system=autotools libs=shared,static arch=linux-debian12-zen4
[+]      ^fftw@3.3.10%gcc@13.3.0+mpi+openmp~pfft_patches+shared build_system=autotools patches=872cff9 precision=double,float arch=linux-debian12-zen4
[+]      ^gcc-runtime@13.3.0%gcc@13.3.0 build_system=generic arch=linux-debian12-zen4
[e]      ^glibc@2.36%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]      ^gmake@4.4.1%gcc@13.3.0~guile build_system=generic arch=linux-debian12-zen4
[+]      ^hwloc@2.11.1%gcc@13.3.0~cairo~cuda~gl~libudev+libxml2~nvml~oneapi-level-zero~opencl+pci~rocm build_system=autotools libs=shared,static arch=linux-debian12-zen4
[+]          ^libpciaccess@0.17%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]              ^util-macros@1.20.1%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]          ^libxml2@2.10.3%gcc@13.3.0+pic~python+shared build_system=autotools arch=linux-debian12-zen4
[+]              ^libiconv@1.17%gcc@13.3.0 build_system=autotools libs=shared,static arch=linux-debian12-zen4
[+]      ^openblas@0.3.28%gcc@13.3.0~bignuma~consistent_fpcsr+dynamic_dispatch+fortran~ilp64+locking+pic+shared build_system=makefile patches=d0b9276 symbol_suffix=none threads=openmp arch=linux-debian12-zen4
[+]      ^openmpi@5.0.5%gcc@13.3.0+atomics~cuda~debug~gpfs~internal-hwloc~internal-libevent~internal-pmix~java~lustre~memchecker~openshmem~romio+rsh~static+vt+wrapper-rpath build_system=autotools fabrics=auto romio-filesystem=none schedulers=none arch=linux-debian12-zen4
[+]          ^autoconf@2.72%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]              ^m4@1.4.19%gcc@13.3.0+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-debian12-zen4
[+]                  ^libsigsegv@2.14%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]          ^automake@1.16.5%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]          ^libevent@2.1.12%gcc@13.3.0+openssl build_system=autotools arch=linux-debian12-zen4
[+]          ^libtool@2.4.7%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]              ^findutils@4.9.0%gcc@13.3.0 build_system=autotools patches=440b954 arch=linux-debian12-zen4
[+]          ^numactl@2.0.18%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]          ^openssh@9.8p1%gcc@13.3.0+gssapi build_system=autotools arch=linux-debian12-zen4
[+]              ^krb5@1.21.3%gcc@13.3.0+shared build_system=autotools arch=linux-debian12-zen4
[+]                  ^bison@3.8.2%gcc@13.3.0~color build_system=autotools arch=linux-debian12-zen4
[+]              ^libedit@3.1-20240808%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
[+]          ^perl@5.40.0%gcc@13.3.0+cpanm+opcode+open+shared+threads build_system=generic arch=linux-debian12-zen4
[+]              ^berkeley-db@18.1.40%gcc@13.3.0+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-debian12-zen4
[+]          ^pmix@5.0.3%gcc@13.3.0~munge~python~restful build_system=autotools arch=linux-debian12-zen4
[+]      ^pkgconf@2.2.0%gcc@13.3.0 build_system=autotools arch=linux-debian12-zen4
</pre></details>
but for ```hsa-rocr-dev-6.2.1``` it gives
```
clang-18: error: cannot find ROCm device library for gfx1200; provide its path via '--rocm-path' or '--rocm-device-lib-path', or pass '-nogpulib' to build without ROCm device library
clang-18: error: cannot find ROCm device library for gfx1201; provide its path via '--rocm-path' or '--rocm-device-lib-path', or pass '-nogpulib' to build without ROCm device library
```

for ```llvm-10.0.1``` it gives
```
cc1plus: note: unrecognized command-line option '-Wno-vla-extension' may have been intended to silence earlier diagnostics
cc1plus: note: unrecognized command-line option '-Wno-deprecated-register' may have been intended to silence earlier diagnostics
ninja: build stopped: subcommand failed.
```

--- a/var/spack/repos/builtin/packages/gromacs/package.py
+++ b/var/spack/repos/builtin/packages/gromacs/package.py
@@ -116,7 +116,7 @@ class Gromacs(CMakePackage, CudaPackage):
         description="Enable multi-GPU FFT support with HeFFTe",
     )
     variant("opencl", default=False, description="Enable OpenCL support")
-    variant("sycl", default=False, when="@2021:", description="Enable SYCL support")
+    variant("sycl", default=False, when="@2021: %clang", description="Enable SYCL support")
     variant(
         "intel-data-center-gpu-max",
         default=False,

